app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -0,0 +1,95 @@
+require 'net/http'
+require 'net/https'
+require 'json'
+
+module Authentication
+  module AuthnIam
+    class Authenticator
+      
+      def initialize(env:)
+        @env = env
+      end
+
+      def valid?(input)
+        @account, @login, password = input.account, input.username, input.password
+
+        # JSON holding the AWS signed headers 
+        signed_aws_headers = JSON.parse password
+
+        response_hash = identity_response_hash(signed_aws_headers)
+        trusted = response_hash != false
+
+        (trusted) && (iam_role_matches? host_role, response_hash)
+
+      end
+
+      def identity_response_hash(signed_aws_headers)
+
+        res = retrieve_iam_identity(signed_aws_headers)
+
+        Rails.logger.info("****> #{res.code} #{res.message}")
+        Rails.logger.info("**** Body -> #{res.body} ")
+
+        if res.code.eql?("200")
```

Comment:
It's not obvious that `code` will be a string, so consider comparing `code.to_i == 200`, or perhaps `code.to_i < 300` (since HTTP statuses such as 204 also indicate success).